components/psst/browser/core/psst_rule_registry_unittest.cc

```diff
@@ -0,0 +1,235 @@
+  if (!success || contents.empty()) {
+    VLOG(2) << "ReadFile: cannot " << "read file " << file_path;
+  }
+  return contents;
+}
+}  // namespace
+
+using OnRuleMatchedCallback = base::RepeatingCallback<void(
+    const std::optional<MatchedRule>& matched_rule)>;
+
+class PsstRuleRegistryUnitTest : public testing::Test {
+ public:
+  void SetUp() override {
+    scoped_feature_list_.InitAndEnableFeature(features::kEnablePsst);
+
+    base::FilePath test_data_dir(
+        base::PathService::CheckedGet(brave::DIR_TEST_DATA));
+    test_data_dir_base_ = test_data_dir.AppendASCII("psst-component-data");
+  }
+
+  base::FilePath GetTestDataDirBase() const { return test_data_dir_base_; }
+  base::FilePath GetScriptsTestDataDir() const {
+    return GetTestDataDirBase().Append(
+        base::FilePath::FromUTF8Unsafe("scripts"));
+  }
+
+ private:
+  content::BrowserTaskEnvironment task_environment_;
+  base::test::ScopedFeatureList scoped_feature_list_;
+  base::FilePath test_data_dir_base_;
+};
+
+TEST_F(PsstRuleRegistryUnitTest, LoadConcreteRule) {
+  PsstRuleRegistry::GetInstance()->ResetRuleRegistryForTest();
+  {
+    base::MockCallback<base::OnceCallback<void(const std::string& data)>>
+        mock_callback;
+    PsstRuleRegistry::GetInstance()->SetOnLoadCallbackForTest(
+        mock_callback.Get());
+
+    base::RunLoop run_loop;
+    EXPECT_CALL(mock_callback, Run)
+        .Times(1)
+        .WillOnce([&](const std::string& data) {
+          EXPECT_EQ(data,
+                    ReadFile(GetTestDataDirBase().Append(
+                        base::FilePath::FromUTF8Unsafe(kPsstJsonFileName))));
+          run_loop.Quit();
+        });
+
+    PsstRuleRegistry::GetInstance()->LoadRules(GetTestDataDirBase());
+    run_loop.Run();
+  }
+
+  const auto scripts_path =
+      GetScriptsTestDataDir().Append(base::FilePath::FromUTF8Unsafe("a"));
+
+  base::RunLoop run_loop;
+  base::MockCallback<
+      base::OnceCallback<void(const std::optional<MatchedRule>&)>>
+      mock_callback;
+  EXPECT_CALL(mock_callback, Run)
+      .Times(1)
+      .WillOnce([&](const std::optional<MatchedRule>& matched_rule) {
+        ASSERT_TRUE(matched_rule);
+        EXPECT_EQ(matched_rule->Name(), "a");
+        EXPECT_EQ(matched_rule->UserScript(),
+                  ReadFile(scripts_path.Append(
+                      base::FilePath::FromUTF8Unsafe(kPsstUserScriptName))));
+        EXPECT_EQ(matched_rule->PolicyScript(),
+                  ReadFile(scripts_path.Append(
+                      base::FilePath::FromUTF8Unsafe(kPsstPolicyScriptName))));
+        run_loop.Quit();
+      });
+
+  PsstRuleRegistry::GetInstance()->CheckIfMatch(GURL("https://a.com"),
+                                                mock_callback.Get());
+  run_loop.Run();
+  ASSERT_EQ(PsstRuleRegistry::GetInstance()->rules_.size(),
+            kTestPsstRulesCount);
+  ASSERT_EQ(PsstRuleRegistry::GetInstance()->component_path_,
```

Comment:
we shouldn't be testing this directly as it's not part of the PsstRuleRegistry contract. It gets tested by calling `CheckIfMatch`